### PR TITLE
Rename acme to e3sm everywhere

### PR DESCRIPTION
**Pull request by rljacob**
Rename all instances of "acme" to "e3sm".

This includes:  the config directory, all function and program names, any "if model = acme" string in python and Fortran.

What was not converted:  references to URLs, file paths and queues with "acme".   References to the cdash dashboard groups.  Some files in docs.  The code in scripts/Tools/pointclm

Testing:  scripts_regression passed.
Test status: bit-for-bit

User interface changes?:  Y   Some Tools with "acme" in the name now say "e3sm".

Update gh-pages html?: N

### CODE REVIEW

File: scripts/Tools/tempgrid.xml [hunk 1]
```diff
@@ -0,0 +1,336 @@
+<?xml version="1.0"?>
+<config_pes>
```
Comment (jgfouca) [inline scripts/Tools/tempgrid.xml]: What's this new file?

File: scripts/Tools/badgrid.xml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0"?>
```
Comment (jedwards4b) [inline scripts/Tools/badgrid.xml]: Why is this file here?

**Comment by rljacob** [comment]
I thought I started with a clean branch with no untracked files but some stuff got swept up in my "git add".  I'll fix it.

**Comment by rljacob** [comment]
regression testing has this error:
```
test_save_timings (__main__.L_TestSaveTimings) ... FAIL
Detected failed test or user request no teardown
Leaving files:
 /lcrc/group/acme/jacob/acme_scratch/scripts_regression_test.20180214_125640/cs.status.fake_testing_only_20180214_130807
 /lcrc/group/acme/jacob/acme_scratch/scripts_regression_test.20180214_125640/SMS_Ln9_P1.f19_g16_rx1.A.anvil_intel.fake_testing_only_20180214_130807

======================================================================
FAIL: test_save_timings (__main__.L_TestSaveTimings)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./scripts_regression_tests.py", line 1940, in test_save_timings
    self.simple_test()
  File "./scripts_regression_tests.py", line 1917, in simple_test
    self._create_test(["SMS_Ln9_P1.f19_g16_rx1.A", timing_flag, "--walltime=0:15:00"], test_id=self._baseline_name)
  File "./scripts_regression_tests.py", line 927, in _create_test
    self._wait_for_tests(test_id, expect_works=(not pre_run_errors and not run_errors))
  File "./scripts_regression_tests.py", line 936, in _wait_for_tests
    from_dir=self._testroot, expected_stat=expected_stat)
  File "./scripts_regression_tests.py", line 61, in run_cmd_assert_result
    test_obj.assertEqual(stat, expected_stat, msg=msg)
AssertionError: 
COMMAND: /lcrc/group/earthscience/jacob/cimeESMCI/scripts/Tools/wait_for_tests  *fake_testing_only_20180214_130807/TestStatus
FROM_DIR: /lcrc/group/acme/jacob/acme_scratch/scripts_regression_test.20180214_125640
SHOULD HAVE WORKED, INSTEAD GOT STAT 100
OUTPUT: Test 'SMS_Ln9_P1.f19_g16_rx1.A.anvil_intel' finished with status 'FAIL'
    Path: SMS_Ln9_P1.f19_g16_rx1.A.anvil_intel.fake_testing_only_20180214_130807/TestStatus
ERRPUT: 

----------------------------------------------------------------------
Ran 94 tests in 922.269s

FAILED (failures=1, skipped=1)
('Detected failures, leaving directory:', '/lcrc/group/acme/jacob/acme_scratch/scripts_regression_test.20180214_125640')
Exit 1

```

Any ideas?

**Comment by jedwards4b** [comment]
look in /lcrc/group/acme/jacob/acme_scratch/scripts_regression_test.20180214_125640/SMS_Ln9_P1.f19_g16_rx1.A.anvil_intel.fake_testing_only_20180214_130807/TestStatus.log

**Comment by jgfouca** [comment]
@rljacob , I'm not 100% sure this test was passing on master an anvil, you may want to confirm that it is before sinking a lot of time into debuggung.

**Comment by rljacob** [comment]
Yeah when I ran on master and didn't use --failfast I only saw this:
```
test_create_test_longname (__main__.K_TestCimeCase) ... ok
test_env_loading (__main__.K_TestCimeCase) ... skipped 'Skipping env load test - Only works on melvin'
Detected failed test or user request no teardown
Leaving files:
test_save_timings (__main__.L_TestSaveTimings) ... ok
```

However it fails later on master. See #2297

**Comment by jgfouca** [comment]
@rljacob , I don't think the TestSaveTimings test uses query_config. Can you look at the case left behind by scripts_regression_tests and find the error?

**Comment by rljacob** [comment]
There's a test later on that **does** use query_config and script-regression fails there on master.
I have no idea where to find all the output from scripts-regression.

**Comment by rljacob** [comment]
Meanwhile, in testing the branch, I'm seeing fail like this:
```
test_run_restart (__main__.T_TestRunRestart) ... FAIL
Detected failed test or user request no teardown
Leaving files:
 /lcrc/group/acme/jacob/acme_scratch/scripts_regression_test.20180215_105835/cs.status.fake_testing_only_20180215_112525
 /lcrc/group/acme/jacob/acme_scratch/scripts_regression_test.20180215_105835/NODEFAIL_P1.f09_g16.X.anvil_intel.fake_testing_only_20180215_112525
```

When I look in the TestStatus.Log, I see:

```
2018-02-15 11:29:19: ERROR: RUN FAIL: Command 'mpiexec  -l  -n 1  /lcrc/group/acme/jacob/acme_scratch/scripts_regression_test.20180215_105835/NODEFA
IL_P1.f09_g16.X.anvil_intel.fake_testing_only_20180215_112525/bld/acme.exe  >> acme.log.$LID 2>&1 ' failed
```

Its failing because there is no "acme.exe".  Its now called "e3sm.exe".   But where is the test getting that executable name from?

**Comment by jedwards4b** [comment]
I told you where!?? - look in /lcrc/group/acme/jacob/acme_scratch/scripts_regression_test.20180214_125640/SMS_Ln9_P1.f19_g16_rx1.A.anvil_intel.fake_testing_only_20180214_130807/TestStatus.log

**Comment by jgfouca** [comment]
@rljacob , if you do this:

./scripts_regression_test RunTests.L_TestSaveTimings

It will fail and the script should tell you the cases it left behind in the last few lines of output.

**Comment by rljacob** [comment]
Ok but I'm ignoring TestSaveTimings for this branch since it DOES fail on master too.

**Comment by jedwards4b** [comment]
Last couple of lines of config_machines.xml 
<default_run_exe>${EXEROOT}/acme.exe </default_run_exe>
  <default_run_misc_suffix> >> acme.log.$LID 2>&amp;1 </default_run_misc_suffix>

**Comment by rljacob** [comment]
Thanks!   Why hardcode the executable name there?

**Comment by rljacob** [comment]
Here's the result before fixing the default run exe:
```
----------------------------------------------------------------------
Ran 126 tests in 2647.905s

FAILED (failures=10, errors=1, skipped=4)
('Detected failures, leaving directory:', '/lcrc/group/acme/jacob/acme_scratch/scripts_regression_test.20180215_105835')
```

**Comment by jedwards4b** [comment]
It's either there or someplace in the code - we should probably remove the other places where exe name is hardcoded and use that one exclusively.

**Comment by jgfouca** [comment]
@rljacob , what is cime model being set to (utils.get_model)?

**Comment by rljacob** [comment]
e3sm.   Doesn't the rest of the system gets the name from MODEL in config_files.xml ?

**Comment by rljacob** [comment]
Or is it CIME_MODEL in namelist_definition_drv.xml ?

**Comment by rljacob** [comment]
Since I just went through the code looking for "acme", I'm pretty confident that's the only place.  The build system references the executable as CIME_MODEL.exe and the runtime system also uses that and open's CIME_MODEL.log so why does scripts-regression need those to be hard-coded?

**Comment by jgfouca** [comment]
@rljacob , how is scripts_regression_tests hardcoded?

**Comment by rljacob** [comment]
Its hardcoding executable and logfile names in the "default_run_suffix" block in config_machines.xml

**Comment by jgfouca** [comment]
@rljacob , that hardcoding was global, not specific to scripts_regression_tests.

**Comment by rljacob** [comment]
Oh I see.  That's where I first noticed it.  So the create_test system also uses that?   Still not clear why its necessary.

**Comment by jgfouca** [comment]
@rljacob , yes, i think several clients were using it. If you update it, things should start working. The root cause for the need of default_run_suffix is legacy from way before cime5. Maybe @jedwards4b can remember why it's needed.

**Comment by rljacob** [comment]
Yes fixing that produced much better test results.
```
----------------------------------------------------------------------
Ran 126 tests in 3556.757s

FAILED (errors=1, skipped=4)
('Detected failures, leaving directory:', '/lcrc/group/acme/jacob/acme_scratch/scripts_regression_test.20180215_115433')
Exit 1
```
the one fail was test_query_components.   I think this is ready for re-review and merging.

File: config/e3sm/machines/config_machines.xml [hunk 3]
```diff
@@ -2837,8 +2837,8 @@
 </machine>
 
 <default_run_suffix>
-  <default_run_exe>${EXEROOT}/acme.exe </default_run_exe>
-  <default_run_misc_suffix> >> acme.log.$LID 2>&amp;1 </default_run_misc_suffix>
+  <default_run_exe>${EXEROOT}/e3sm.exe </default_run_exe>
```
Comment (jedwards4b) [inline config/e3sm/machines/config_machines.xml]: I don't remember why this was separate.  I think you could do ${EXEROOT}/${MODEL}.exe and
${MODEL}.log.$LID

Comment (rljacob) [reply config/e3sm/machines/config_machines.xml]: I'd rather do that in a separate PR.

**Comment by rljacob** [comment]
@jgfouca what is your plan for merging this to ESMCI and then E3SM?

**Comment by jgfouca** [comment]
@rljacob , we'll see how the E3 merge goes next week.

**Comment by rljacob** [comment]
Ok.  I'd like this to go in kind of by itself or at least without any other big changes.
